### DISCUSSION TITLE
Support for in-app purchase or other payment methods

### DISCUSSION THREAD

**Discussion by wjkoh** [Instawork/hyperview#845] [Q&A] [answered]
Hello everyone,

we are considering converting our mobile app to a Hyperview app. Before dedicating resources to this project, I would like to inquire about Hyperview's capabilities for handling in-app purchases or other payment methods approved by the Apple app store. I have been unable to find any relevant documentation on this topic. Can anyone provide me with information or advice on this matter?

Thank you.

**Comment by adamstep** [comment, marked as answer]
Hi @wjkoh , Hyperview does not have support for IAP or other payment methods out of the box. Support for those would need to be added via extensions (custom behaviors and elements). How difficult this will be depends on the answer to the question: is your current mobile app written in React Native already?

If the answer is yes, you could expose your current IAP flows rather easily (custom behavior to trigger a purchase or subscription with a given product ID, and then trigger an event on success/failure). I think this would be rather straight-forward. Another option is to write most of the app in Hyperview, but keep the screens that deal with purchases as React Native implementations.

If your current app is not written in React Native, then you can expose the functionality in Hyperview by using the [React Native IAP](https://react-native-iap.dooboolab.com/docs/get-started/) library and wrapping that as extensions. I don't have experience with IAP myself, but my understanding is that the flows are quite complex and take a lot of time to set up properly ([see this diagram](https://react-native-iap.dooboolab.com/docs/guides/purchases#new-purchase-flow)). So please keep that in mind when considering if converting the entire app is right for you at this time.

**Reply by wjkoh** [reply]
Thank you for your response, @adamstep. Our mobile app is built with Flutter, so migrating to React Native would not be a simple copy-and-paste process for us. However, as you mentioned, we can enhance the Hyperview client with custom React Native code, allowing us to achieve the same functionality as any React Native client. This gives us the assurance we need to proceed with Hyperview. While we do not have much experience with React Native, it appears to be feasible based on our review of the IAP library. Thank you again for your response, and I will return to this forum if we encounter any significant challenges during the migration process.

**Reply by adamstep** [reply]
Sounds good, yes please reach out if you have questions or want some feedback on how to do the integration. Good luck!
